### ISSUE TITLE
New Cutter needs thorough testing

### ISSUE THREAD

**Issue by scottkleinman** [WheatonCS/Lexos#401]
Forked from issue #398.

The new ajax version of Cutter seems to be working, but there are a lot of possible combinations of options to test. I'm leaving this issue open until we're sure it's stable.

**Comment by mleblanc321** [comment]
Lexos hanging 
in (new) cutter
ran DAZ (tried left-side options for AZ   1 segment
and Individual Options on Daniel (hmm, are individual options working?)

twirling icon showing ...

**Comment by evanlaferriere** [comment]
GenAB is working, meaning tokens/segment and scrubber are working. However, I still cannot access individual options on cutter.

**Comment by evanlaferriere** [comment]
Do we want cut by milestone to be the only option that brings up a new screen?

**Comment by scottkleinman** [comment]
Just catching up on this morning's issues. I pushed a fix for the individual options buttons last night. So you may need to pull again. They are working for me in Firefox and Chrome.

If you switch from tokens/segment to segments/document, you will notice that some options are greyed out. We could do that with the milestones checkbox, greying out the other options when it is checked.

**Comment by scottkleinman** [comment]
Damn! The individual options buttons were working when I went to bed, and now they're not. I'll look into this.

**Comment by evanlaferriere** [comment]
I've been testing out how it responds to the user having no documents, the only time a dialog informing them of no documents comes up when you're in cut by milestone and have entered a string to cut by. In the other options, the default cutting error modal pops up whenever there is nothing entered in that input, and when there is something there, the invalid overlap error appears and the user will never actually receive the "no documents" message.

**Comment by scottkleinman** [comment]
That doesn't entirely surprise me. It was hard to translate the logic embedded in the old template into javascript. That probably just needs adjusting. Unfortunately, I am having a horrible time debugging this, as the browser keeps caching the old code, and nothing I do seems to get rid of it. I'll update you with progress as soon as I can.

**Comment by evanlaferriere** [comment]
Which browser are you using?

**Comment by scottkleinman** [comment]
I just pushed a fix for the Individual Cutting Options buttons. Tested in Firefox. I had to go into the browser options and clear the cache there after making each change. Very annoying!

**Comment by SocialHulkHogan** [comment]
Scott see issue 403. Something broke wordcloud and bubbleviz. I think it's the error modal code. I can't figure out what broke them

**Comment by SocialHulkHogan** [comment]
those pages currently do nothing

**Comment by evanlaferriere** [comment]
Can confirm for chrome, browser was caching in such a way to prevent individual options from doing anything when clicked. When I cleared  my cache in chrome, it started working. Firefox and opera are also working fine.

**Comment by evanlaferriere** [comment]
Just tried to run DAZ, cutter froze

**Comment by evanlaferriere** [comment]
I just added an extra else in the error trap section in the doScrubbing js function to fix the no documents error. The other error windows would pop up and replace it since they were not elsed out.

**Comment by evanlaferriere** [comment]
The caching error came back, and was fixed again by clearing it. What's causing this?

**Comment by scottkleinman** [comment]
No idea, but it's effecting my work on other pages as well.

**Comment by evanlaferriere** [comment]
Entering floats as the segment size seems like it should be caught with err4 "Invalid segment size" but it seems like it tries to cut but can't and just loads.

**Comment by evanlaferriere** [comment]
CORRECTION: Clearing the cache solves this problem.

**Comment by evanlaferriere** [comment]
The only error I could not get is err6 "Individual cutting: Invalid Segment size", which gets a "Lexos could not apply scrubbing actions".

**Comment by evanlaferriere** [comment]
The individual options css gets changed after a successful run of the cut tool.

**Comment by evanlaferriere** [comment]
However, I have been able to use individual options successfully. If they are not working, I am able to just refresh the page, and after doing that once or twice, their old CSS is restored and they will be usable. Once cutter has been executed, the individual options lose their styling, and the buttons are worthless and will leave you loading forever.

**Comment by evanlaferriere** [comment]
I was able to replicate DAZ using the individual options

**Comment by scottkleinman** [comment]
This is all good news. I'm aware of the css change. Unfortunately, there needs to be separate implementations of the toggle mechanism, depending on whether you've just loaded the page or whether you've just clicked apply. I left the css difference in there so that we can tell the difference. When everything else is working, I'll fix that.

**Comment by evanlaferriere** [comment]
Firefox seems to be working as well chrome

**Comment by scottkleinman** [comment]
I feel like we've covered the technical problems in this issue except for the error handling logic. I'm going to close this issue and open a new one for that.

**Comment by evanlaferriere** [comment]
I just pulled and I'm experiencing the bug where it just loads forever, no errors, and all that's in the terminal is
```
Wiping session (5N3N3P6TH2JV1NP38RWLGLIQXJB2H5) and old files...
Attempting new id of 1U75C00AN9SELW6NI0UTO7N0R095A0 ... Good.
Initialized new session, session folder, and empty file manager with id.
127.0.0.1 - - [29/Jun/2016 10:02:31] "GET /upload HTTP/1.1" 200 -
127.0.0.1 - - [29/Jun/2016 10:02:33] "POST /upload HTTP/1.1" 200 -
127.0.0.1 - - [29/Jun/2016 10:02:36] "GET /cut HTTP/1.1" 200 -
```

**Comment by mleblanc321** [comment]
ugh, yup, chase this

**Comment by evanlaferriere** [comment]
I was just running the DAZ experiment again when it happened. It was happening for a few minutes but now it seems to be working again on the same settings. I did not clear the cache, I only refreshed the page. I'm unsure how to reproduce the bug.

**Comment by evanlaferriere** [comment]
With several thousand docs, ITM will not close on the cut page. If you click one of the action buttons, it freezes the page.

**Comment by scottkleinman** [comment]
I'm wondering if the caching issue we have somehow relates to how fast the javascript libraries are loaded from CDN. Do things go haywire if there's a blip?

"Individual options overlap input should default to 0 if left empty...". This seems to happen only after you have previewed. I'll have a look at it. As for the separate styling of the button, I'll change that later today.

"With several thousand docs..." Evan, can you describe this scenario further? Did you cut into several thousand segments and then reload the page?

**Comment by evanlaferriere** [comment]
It seems to me that whatever was causing the bug that reopened this was doCutting() not being called. It happens infrequently, and I'm not sure what causes it so I cannot replicate it when it does happen. Refreshing fixes it. It seems to me that the ajax is failing, so the "Lexos could not apply the cutting actions" error should be thrown, but it isn't.  
In the case of several thousand I've been splicing Genesis over and over to try and break cutter in a new way. It actually is working, it just takes really, really long. The ITM takes over half the page until it loads.

**Comment by scottkleinman** [comment]
OK, the ITM thing is an open issue (issue #361). I think we'll have to tackle that separately, as it applies to any page that takes a long time to load.

For the first problem, I suppose we should try to catch unexplained errors and show a message advising the user to refresh the page. I'll look into that.

**Comment by evanlaferriere** [comment]
```
Traceback (most recent call last):
  File "/home/lexos/anaconda2/lib/python2.7/site-packages/werkzeug/serving.py", line 193, in run_wsgi
    execute(self.server.app)
  File "/home/lexos/anaconda2/lib/python2.7/site-packages/werkzeug/serving.py", line 184, in execute
    write(data)
  File "/home/lexos/anaconda2/lib/python2.7/site-packages/werkzeug/serving.py", line 152, in write
    self.send_header(key, value)
  File "/home/lexos/anaconda2/lib/python2.7/BaseHTTPServer.py", line 401, in send_header
    self.wfile.write("%s: %s\r\n" % (keyword, value))
IOError: [Errno 32] Broken pipe
```
This is what happens when you have >10,000 docs

**Comment by scottkleinman** [comment]
Whoa! That's helpful. It's a back-end issue. Broken pipes when writing files is new territory for me.

On the front end, this should just trigger the ajax error function. Right now, that just outputs an error to the console. We could easily trigger a modal error message instead.

**Comment by evanlaferriere** [comment]
Just noticed that I'm not getting the "Over 100 new segments" warning when I'm cutting thousands of docs and definitely should be gettting that message.

**Comment by scottkleinman** [comment]
Merged into #447.
